wpilibj/src/main/java/edu/wpi/first/wpilibj/simulation/DCMotorSim.java

```diff
@@ -31,90 +53,182 @@ public class DCMotorSim extends LinearSystemSim<N2, N1, N2> {
+    //   B = GKₜ/(RJ)
+    //   J = GKₜ/(RB)
+    m_gearing = -gearbox.KvRadPerSecPerVolt * plant.getA(1, 1) / plant.getB(1, 0);
+    m_jKgMetersSquared = m_gearing * gearbox.KtNMPerAmp / (gearbox.rOhms * plant.getB(1, 0));
   }
 
   /**
-   * Creates a simulated DC motor mechanism.
+   * Sets the state of the DC motor.
    *
-   * @param gearbox The type of and number of motors in the DC motor gearbox.
-   * @param gearing The gearing of the DC motor (numbers greater than 1 represent reductions).
-   * @param jKgMetersSquared The moment of inertia of the DC motor. If this is unknown, use the
-   *     {@link #DCMotorSim(LinearSystem, DCMotor, double, double...)} constructor.
-   * @param measurementStdDevs The standard deviations of the measurements. Can be omitted if no
-   *     noise is desired. If present must have 2 elements. The first element is for position. The
-   *     second element is for velocity.
+   * @param angularPositionRad The new position in radians.
+   * @param angularVelocityRadPerSec The new velocity in radians per second.
+   * @return this simulated DC motor with applied state.
    */
-  public DCMotorSim(
-      DCMotor gearbox, double gearing, double jKgMetersSquared, double... measurementStdDevs) {
-    super(
-        LinearSystemId.createDCMotorSystem(gearbox, jKgMetersSquared, gearing), measurementStdDevs);
-    m_gearbox = gearbox;
-    m_gearing = gearing;
+  public DCMotorSim setState(double angularPositionRad, double angularVelocityRadPerSec) {
+    setState(VecBuilder.fill(angularPositionRad, angularVelocityRadPerSec));
+    return this;
   }
 
   /**
-   * Sets the state of the DC motor.
+   * Sets the DC motor's angular position.
    *
    * @param angularPositionRad The new position in radians.
+   * @return this simulated DC motor with applied position.
+   */
+  public DCMotorSim setAngle(double angularPositionRad) {
+    return setState(angularPositionRad, getAngularVelocityRadPerSec());
+  }
+
+  /**
+   * Sets the DC motor's angular velocity.
+   *
    * @param angularVelocityRadPerSec The new velocity in radians per second.
+   * @return this simulated DC motor with applied position.
    */
-  public void setState(double angularPositionRad, double angularVelocityRadPerSec) {
-    setState(VecBuilder.fill(angularPositionRad, angularVelocityRadPerSec));
+  public DCMotorSim setAngularVelocity(double angularVelocityRadPerSec) {
+    return setState(getAngularPositionRad(), angularVelocityRadPerSec);
+  }
+
+  /**
+   * Returns the gear ratio of the DC motor.
+   *
+   * @return the DC motor's gear ratio.
+   */
+  public double getGearing() {
+    return m_gearing;
+  }
+
+  /**
+   * Returns the moment of inertia of the DC motor.
+   *
+   * @return The DC motor's moment of inertia.
+   */
+  public double getJKgMetersSquared() {
+    return m_jKgMetersSquared;
+  }
+
+  /**
+   * Returns the gearbox for the DC motor.
+   *
+   * @return The DC motor's gearbox.
+   */
+  public DCMotor getGearBox() {
```

Comment:
We use "Gearbox" everywhere else.
```suggestion
  public DCMotor getGearbox() {
```
There's also one existing instance of this misspelling in FlywheelSim.java.